src/clients/Transmission.ts

```diff
@@ -170,26 +186,44 @@ export default class Transmission implements TorrentClient {
 			downloadDir = result.unwrapOrThrow().downloadDir;
 		}
 
-		let addResponse: TorrentAddResponse;
-
 		try {
-			addResponse = await this.request<TorrentAddResponse>(
+			const addResponse = await this.request<TorrentAddResponse>(
 				"torrent-add",
 				{
 					"download-dir": downloadDir,
 					metainfo: newTorrent.encode().toString("base64"),
-					paused: false,
+					paused: true,
 					labels: ["cross-seed"],
 				}
 			);
+			if (doesAlreadyExist(addResponse)) {
+				return InjectionResult.ALREADY_EXISTS;
+			}
+
+			const torrentId = addResponse["torrent-added"].id;
+
+			const getResponse = await this.request<TorrentGetResponseArgs>(
+				"torrent-get",
+				{
+					ids: [torrentId],
+					fields: ["percentDone", "status"],
+				}
+			);
```

Comment:
The whole torrent recheck is definitely not synchronous, as the app just kicks it off if it sees an issue with the file, but doesn't wait for the full re-check to complete (obviously, or the API call would just be hanging):
https://github.com/transmission/transmission/blob/fd4d097d13ab4236550fa01d411fd63fc161f38b/libtransmission/torrent.cc#L1002

If you're asking if:
- checking the logic to see if a re-seed is complete (i.e. essentially if all files are there and the first chunk passes the checksum test)
- if yes, set the torrent as complete, and if not, trigger a reckeck
- change the status of the torrent so that torrent-get will return the correct status

Then I don't know, honestly. I have never visually seen it take any time at all, as far as I can tell it is absolutely instant. But that's all I have, as the Transmission codebase is large.